### ISSUE TITLE
Create a function to find a piece on the board

### ISSUE THREAD

**Issue by weerox** [weerox/rip-lib#9]

**Comment by default-username-852** [comment]
I don't know exactly what you want changed, but pull request #11 might resolve this issue.

**Comment by weerox** [comment]
I actually already implemented this, but didn't have a change to commit and push. 

What I meant was to create a function, which takes a piece as a parameter and find all pieces on the board which matches that piece.

**Comment by Nilslof** [comment]
i didn't actually see that this one was already compleated, but i wrote a code to find a piece either from a color and name o from anouther piece.

```
fn find_piece(&mut self, name: Name, color:Color) -> Vec<(Rank, File)>{
	let mut vector: Vec<(Rank, File)> = Vec::new();
	for w in 0.. self.width{
		for h in 0..self.height{
			if let Some(piece) = self.board[w][h].piece{
				if piece.color == color && piece.name == name{
					vector.push((self.board[w][h].rank, self.board[w][h].file));
				}
			}
		}
	}
	return vector;
}

fn find_piece_from_piece(&mut self, piece:Piece) -> Vec<(Rank, File)>{
	return self.find_piece(piece.name, piece.color);
}
```

**Comment by weerox** [comment]
@default-username-852 I think that the innermost if-clause in the code-snippet above is the correct way to check if two pieces is the same (unless a equality trait is implemented in Piece, but that seems unnecessary).

**Comment by Nilslof** [comment]
@weerox well since both traits Derives PartialEq, this should work

**Comment by weerox** [comment]
Yes, your solution would work, but the original PR by @default-username-852 made an equality check between two pieces, which (probably, I'm not sure) would have also made an equality check on the `moved` boolean, but in this context, two pieces are equal regardless of whether one of them has moved or not.

**Comment by weerox** [comment]
The PR has been fixed though.

**Comment by weerox** [comment]
Fixed by #13
